src/components/warnings.ts

```diff
@@ -11,12 +11,14 @@ export type WarningsPanelOptions = {
 		onClickWaning: (warning: Warning) => void;
 	};
 };
+
 /** Setup a component to display warnings. */
 export function setupWarningsPanel({ element, listeners }: WarningsPanelOptions) {
 	return {
 		setResult: (result: LinterServiceResult) => {
```

Comment:
nit
refactor

`({ result, returnCode, ruleMetadata }: LinterServiceResult) =>`
i.e. to avoid confusing instances of `result.result`

Reply:
The code would be simpler, but I want to keep it in this PR to minimize diffs.